test/src/main/java/org/apache/accumulo/test/upgrade/GCUpgrade9to10TestIT.java

```diff
@@ -118,38 +113,70 @@ public void gcUpgradeNoDeletesIT() throws Exception {
   }
 
   /**
-   * This is really hard to make happen - the minicluster can only use so little memory to start up.
-   * The {@link org.apache.accumulo.master.upgrade.Upgrader9to10} CANDIDATE_MEMORY_PERCENTAGE can be
-   * adjusted.
+   * Ensure that the size of the candidates exceeds the {@link Upgrader9to10}'s CANDIDATE_BATCH_SIZE
+   * and will clean up candidates in multiple batches, without running out of memory.
    */
   @Test
   public void gcUpgradeOutofMemoryTest() throws Exception {
     killMacGc(); // we do not want anything deleted
 
-    int somebignumber = 100000;
-    String longpathname = "aaaaaaaaaabbbbbbbbbbccccccccccddddddddddeeeeeeeeee"
-        + "ffffffffffgggggggggghhhhhhhhhhiiiiiiiiiijjjjjjjjjj"
-        + "kkkkkkkkkkkkkkkkkklllllllllllllllllllllmmmmmmmmmmmmmmmmmnnnnnnnnnnnnnnnn";
-    longpathname += longpathname; // make it even longer
+    int numberOfEntries = 100_000;
+    String longpathname = StringUtils.repeat("abcde", 100);
+    assertEquals(500, longpathname.length());
+
+    // dynamically get the batch size from the other class
+    Field CANDIDATE_BATCH_SIZE_field = Upgrader9to10.class.getDeclaredField("CANDIDATE_BATCH_SIZE");
```

Comment:
Could make this field public.

Reply:
Yes. I'm not sure why I didn't. Maybe just trying to limit what was exposed. I can make that change.